dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/handlers/DimensionDataCloudControlErrorHandler.java

```diff
@@ -47,16 +47,20 @@ public void handleError(HttpCommand command, HttpResponse response) {
             String.format("%s -> %s", command.getCurrentRequest().getRequestLine(), response.getStatusLine());
       switch (response.getStatusCode()) {
          case 400:
-            if (message.contains("RESOURCE_NOT_FOUND") || message.contains("OPERATION_NOT_SUPPORTED")) {
+            if (message.contains("RESOURCE_NOT_FOUND")) {
                exception = new ResourceNotFoundException(message, exception);
-            } else if (message.contains("INVALID_INPUT_DATA") || message.contains("ORGANIZATION_NOT_VERIFIED")
-                  || (message.contains("SYSTEM_ERROR") && !message.contains("RETRYABLE_SYSTEM_ERROR")) || message
-                  .contains("CPU_SPEED_NOT_AVAILABLE") || message.contains("CONFIGURATION_NOT_SUPPORTED")) {
+            } else if (message.contains("OPERATION_NOT_SUPPORTED")) {
+               exception = new UnsupportedOperationException(message, exception);
+            } else if (message.contains("RESOURCE_BUSY") || message.contains("RESOURCE_LOCKED")) {
                exception = new IllegalStateException(message, exception);
-            } else if (message.contains("RESOURCE_BUSY") || message.contains("UNEXPECTED_ERROR")) {
-               exception = new ResourceNotFoundException(message, exception);
             } else if (message.contains("NAME_NOT_UNIQUE")) {
                exception = new ResourceAlreadyExistsException(message, exception);
+            } else if (message.contains("UNEXPECTED_ERROR")
+                  || message.contains("RETRYABLE_SYSTEM_ERROR")
+                  || message.contains("SYSTEM_ERROR")) {
+               break;
+            } else {
+               exception = new IllegalArgumentException(message, exception);
```

Comment:
There are too many responseCode value to list here. Rather than that, it is better to default to IllegalArgumentException